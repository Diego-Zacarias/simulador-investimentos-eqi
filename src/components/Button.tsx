import React from "react";
import styled, { css } from "styled-components";

interface StyleProps {
  variant: string;
}
const Root = styled.button<StyleProps>`
  width: 100%;
  max-width: 248px;
  height: 40px;
  margin-top: 32px;
  border: 1px solid #000;
  border-radius: 8px;
  font-weight: bold;
  cursor: pointer;
  ${(props) =>
    props.variant === ButtonVariant.default
      ? css`
          background-color: ${(props) => props.theme.colors.primary};
        `
      : css`
          background-color: transparent;
        `};
  &:disabled {
    background-color: ${(props) => props.theme.colors.disabled};
    color: ${(props) => props.theme.colors.typography};
    cursor: default;
  }
`;

export const ButtonVariant = {
  default: "default",
  outlined: "outlined",
};

export type variantType = "default" | "outlined"

interface ButtonProps {
  children: JSX.Element;
  variant: variantType;
  type?: any;
  handleClick?: any;
  disabled?: boolean;
}

const Button = ({
  children,
  variant = "default",
  handleClick,
  disabled = false,
  type
}: ButtonProps) => {
  return (
    <Root
      variant={variant}
      onClick={handleClick}
      disabled={disabled}
      type={type}
    >
      {children}
    </Root>
  );
};

export default Button;
