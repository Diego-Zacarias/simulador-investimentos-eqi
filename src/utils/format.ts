export const format = (number: number): string => {
  return number.toLocaleString("pt-br", { style: "currency", currency: "BRL" });
};
