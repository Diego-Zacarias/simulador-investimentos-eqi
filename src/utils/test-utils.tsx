import React from "react";
import { render } from "@testing-library/react";
import ThemeProvider from "../styles/ThemeProvider";

const TestProvider = ({ children }: any): JSX.Element => (
      <ThemeProvider>{children}</ThemeProvider>
  );

export * from "@testing-library/react";

const customRender = (ui: any, options?: any) =>
  render(ui, { wrapper: TestProvider, ...options });

export { customRender as render };
