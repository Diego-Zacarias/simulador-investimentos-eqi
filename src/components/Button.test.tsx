import { render, screen, fireEvent } from "../utils/test-utils";

import Button from "./Button";

test('renderiza com o testo "Click aqui"', () => {
  render(
    <Button variant="default">
      <p>Click aqui</p>
    </Button>
  );
  const linkElement = screen.getByText(/Click aqui/i);
  expect(linkElement).toBeInTheDocument();
});

test("Disparar evento ao clicar", async () => {
  const handleClick = jest.fn();
  render(
    <Button variant="default" handleClick={handleClick}>
      <p>Click aqui</p>
    </Button>
  );
  await fireEvent.click(screen.getByRole("button"));
  expect(handleClick).toBeCalled();
});

test.each([["default"], ["outlined"]])("renderiza com a variante %s", 
  (variant: any) => {
    const { asFragment } = render(
      <Button variant={variant}>
        <p>Button</p>
      </Button>
    );

    expect(asFragment()).toMatchSnapshot();
});
