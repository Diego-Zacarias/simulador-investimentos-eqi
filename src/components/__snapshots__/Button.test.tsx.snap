// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderiza com a variante default 1`] = `
<DocumentFragment>
  .c0 {
  width: 100%;
  max-width: 248px;
  height: 40px;
  margin-top: 32px;
  border: 1px solid #000;
  border-radius: 8px;
  font-weight: bold;
  cursor: pointer;
  background-color: #ED8E53;
}

.c0:disabled {
  background-color: #969696;
  color: #1F1F1F;
  cursor: default;
}

<button
    class="c0"
  >
    <p>
      Button
    </p>
  </button>
</DocumentFragment>
`;

exports[`renderiza com a variante outlined 1`] = `
<DocumentFragment>
  .c0 {
  width: 100%;
  max-width: 248px;
  height: 40px;
  margin-top: 32px;
  border: 1px solid #000;
  border-radius: 8px;
  font-weight: bold;
  cursor: pointer;
  background-color: transparent;
}

.c0:disabled {
  background-color: #969696;
  color: #1F1F1F;
  cursor: default;
}

<button
    class="c0"
  >
    <p>
      Button
    </p>
  </button>
</DocumentFragment>
`;
